File: Leetcode/Leetcode-DSA-course/Stacks-and-Queues/739-daily-temperatures/daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        stack<int> stack;
        vector<int> answer(temperatures.size());
        for (int i = 0; i < temperatures.size(); i++) {
            while (!stack.empty() &&
                   temperatures[stack.top()] < temperatures[i]) {
                int j = stack.top();
                answer[j] = i - j;
                stack.pop();
            }
            stack.push(i);
        }
        return answer;
    }
};

File: Leetcode/Leetcode-DSA-course/Stacks-and-Queues/739-daily-temperatures/daily-temperatures.ts
function dailyTemperatures(temperatures: number[]): number[] {
    let stack = []
    let ans = new Array(temperatures.length).fill(0)
    for (let i = 0; i < temperatures.length; i++) {
        while (stack.length > 0 && temperatures[stack[stack.length - 1]] < temperatures[i]) {
            let j = stack[stack.length - 1]
            ans[j] = i - j
            stack.pop()
        }
        stack.push(i)
    }
    return ans;
};